### PR TITLE
fix: add commands for generating go types

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: #13

### CODE REVIEW

File: buf.yaml
Comment (rootulp) [inline buf.yaml:9]: to satisfy yamllint

```suggestion
```

File: Makefile
Comment (rootulp) [inline Makefile:14]: [question] do we want to support two different ways to generate the Protobuf files? I think it easier to maintain just one option and if that option is used in CI as well as locally. Since the `make proto-gen` command depends on a locally installed Buf CLI, preference for the Docker approach.

Ideally the Docker approach uses the same version of Buf locally and in CI.

**Review by rootulp — COMMENTED** [review]
Note: I'm observing that when I run  `make proto-gen-docker` or `make proto-gen`, there are changes to these files:

```
	modified:   merkle/proto/proof.pb.go
	modified:   pkg/blob/blob.pb.go
```

**Review by staheri14 — APPROVED** [review]
LGTM!

File: Makefile
Comment (cmwaters) [reply Makefile:14]: We could probably look to enforce it somehow. The docker image also isn't versioned (I actually had a problem where it wasn't working because I had an old image.

I think some people don't like using docker, hence why I added the other option, but it's simple enough that it probably doesn't need a make command so I will remove it

**Comment by cmwaters** [comment]
> Note: I'm observing that when I run make proto-gen-docker or make proto-gen, there are changes to these files:

We might have different versions - let me dig into it

File: Makefile
Comment (rootulp) [reply Makefile:14]: Yea I think the Docker approach should also be versioned.

**Comment by rootulp** [comment]
If it helps debug:

```
$ buf --version
1.28.1
```

```
docker images

bufbuild/buf                                  latest    5085a683a6ba   3 months ago    55.4MB
```

**Comment by staheri14** [comment]
> Note: I'm observing that when I run `make proto-gen-docker` or `make proto-gen`, there are changes to these files:
> 
> ```
> 	modified:   merkle/proto/proof.pb.go
> 	modified:   pkg/blob/blob.pb.go
> ```

It is the case for me too.

**Comment by cmwaters** [comment]
Okay I had an older version of buf. I'm going to hardcode the version to 1.28.1 in the Makefile so hopefully that produces constant results

**Review by rootulp — APPROVED** [review]
`make proto-gen` results in no changes and `make proto-lint` results in no issues. Thanks!

**Comment by rootulp** [comment]
We likely need a maintainer to make it so 2 approvals are needed in this repo.

**Comment by rootulp** [comment]
Hmm we need to update settings on this repo so that new commits dismiss previous PR approvals.

**Comment by cmwaters** [comment]
> Hmm we need to update settings on this repo so that new commits dismiss previous PR approvals.

Done. I've also made it require two reviews like the other repos

**Review by staheri14 — APPROVED** [review]

**Pull request merged**
